internal/backend/post.go

```diff
@@ -46,64 +48,74 @@ func (d *delphisBackend) CreatePost(ctx context.Context, discussionID string, pa
 		ImportedContentID: input.ImportedContentID,
 	}
 
-	// Begin tx
-	tx, err := d.db.BeginTx(ctx)
-	if err != nil {
-		logrus.WithError(err).Error("failed to begin tx")
-		return nil, err
-	}
-	// Put post contents
-	if err := d.db.PutPostContent(ctx, tx, postContent); err != nil {
-		logrus.WithError(err).Error("failed to PutPostContent")
-
-		// Rollback on errors
-		if txErr := d.db.RollbackTx(ctx, tx); txErr != nil {
-			logrus.WithError(txErr).Error("failed to rollback tx")
-			return nil, multierr.Append(err, txErr)
+	retryAttempts := 0
```

Comment:
This has to be done at backend-level, and not at datastore-level. This is due to the fact that the transactions is managed in the backend, and retrying inside the datastore would always break it.